### PR TITLE
clarify push manifest spec

### PR DESCRIPTION
**Pull request by laflechejonathan**
See discussion at: https://github.com/moby/buildkit/issues/2963

It's not currently obvious from reading the spec that manifests should be persisted in the exact wire format provided by the client, leading to errors if a registry re-formats the manifest.

This code in containerd validates the `Docker-Content-Digest` header after a manifest push, and will throw if they don't match:

https://github.com/moby/buildkit/blob/874eef9b70dbaf4f074d2bc8f4dc64237f8e83a0/vendor/github.com/containerd/containerd/remotes/docker/pusher.go#L418-L425

### CODE REVIEW

**Review by jdolitsky — APPROVED** [review]
@laflechejonathan - could you please fix the DCO issue?

**Review by sudo-bmitch — APPROVED** [review]

**Review by jdolitsky — APPROVED** [review]

**Comment by laflechejonathan** [comment]
@jdolitsky are we good to merge this?

File: spec.md [hunk 1]
```diff
@@ -423,13 +423,17 @@ Manifest byte stream:
 
 The uploaded manifest MUST reference any blobs that make up the artifact.
 However, the list of blobs MAY be empty.
+
+The registry MUST store the manifest in the exact byte representation provided by the client.
```
Comment (mikebrow) [inline spec.md:427]: I might've worded "in the exact byte representation" more like "byte for byte exactly as"

File: spec.md [hunk 2]
```diff
@@ -423,13 +423,17 @@ Manifest byte stream:
...
 Upon a successful upload, the registry MUST return response code `201 Created`, and MUST have the following header:
 
 ```
 Location: <location>
 ```
 
 The `<location>` is a pullable manifest URL.
+The Docker-Content-Digest header returns the canonical digest of the uploaded blob, and MUST be equal to the client provided digest.
```
Comment (mikebrow) [inline spec.md:435]: suggest rewording this to fit the paragraph..

what if the digest differ? What if the client did not provide one..

File: spec.md (hunk 1 above)
Comment (sudo-bmitch) [reply spec.md:427]: Since this one is merged, any wording changes now should be pushed as a new PR.

File: spec.md [hunk 3]
```diff
@@ -423,13 +423,17 @@ Manifest byte stream:
...
+Clients MAY ignore the value but if it is used, the client SHOULD verify the value against the uploaded blob data.
```
Comment (mikebrow) [inline spec.md:436]: manifest?

**Comment by mikebrow** [comment]
see comments questions

File: spec.md (hunk 1 above)
Comment (mikebrow) [reply spec.md:427]: :-)
